### ISSUE TITLE
Broken characters of apidoc

### ISSUE THREAD

**Issue by sxcooler** [yiisoft/yii2-apidoc#67]
### What steps will reproduce the problem?

I use vendor/bin/apidoc api (source dir) (output dir), and I got a set of htmls.
then I found that characters of Chinese comment was broken, but english is OK.
![image](https://cloud.githubusercontent.com/assets/3192045/15463819/61bf66d0-20fe-11e6-871c-33627d5f5ab2.png)
This will occurs with method comment & method params comment, I haven't test them all.
### What's expected?

![image](https://cloud.githubusercontent.com/assets/3192045/15463857/9bc9d130-20fe-11e6-8c00-66a909bba083.png)
### What do you get instead?

![image](https://cloud.githubusercontent.com/assets/3192045/15463889/b4197894-20fe-11e6-9f05-7ccdd6128ae6.png)
### Additional info

| Q | A |
| --- | --- |
| Yii vesion | 2.0.8 |
| PHP version | 5.6.19 |
| Operating system | OS X 10.11.4 |

**Comment by samdark** [comment]
Looks like usage of not `mb_` functions for string manipulation.

**Comment by cebe** [comment]
could you provide the doc comment in text form so I can copy it for testing?

**Comment by sxcooler** [comment]
/**
     * test comment
     *
     * @return array
     */


/**
     * 将几个json字段自动encode
     *
     * @return bool
     */


 /**
     * actionOrder 产品预订页
     * 
     * @param integer $id 注意是productId
     * @return string
     * @throws HttpException
     */


Something like that.


Thank you for your reply.



At 2016-05-23 21:07:42, "Carsten Brandt" <notifications@github.com> wrote:


could you provide the doc comment in text form so I can copy it for testing?

**Comment by cebe** [comment]
I am unable to reproduce this. Are you sure your file is UTF-8 encoded?

**Comment by sxcooler** [comment]
Yes I'm sure, I can see it in the git, in the editor, and my whole application works well with all encoding-settings are utf-8.






At 2016-05-23 22:23:17, "Carsten Brandt" <notifications@github.com> wrote:


I am unable to reproduce this. Are you sure your file is UTF-8 encoded?

**Comment by cebe** [comment]
I am running the following steps in an empty directory:

```
composer require yiisoft/yii2-apidoc
ln -s bower-asset/ vendor/bower
mkdir test
```

create the file `test/SomeClass.php`: 

```php
<?php

/**
 * 将几个json字段自动encode
 */
class SomeClass
{
	/**
	 * 将几个json字段自动encode
	 * @param int $id 将几个json字段自动encode
	 * @return string 将几个json字段自动encode
	 */
	public function test($id)
	{

	}
}

```

and run:

```
./vendor/bin/apidoc api test/ apidocs
```

the output is fine. No idea how to reproduce it.

**Comment by sxcooler** [comment]
Thank u for all these, I'll debug it tomorrow and let u know if anything Found.

发自我的 iPhone

**Comment by sxcooler** [comment]
I debugged this step by step and I found this:
in vendor/yiisoft/yii2-apidoc/models/BaseDoc.php line106


yii2 used ucfirst() on the shortDescription, but it’s not safe with non-latin characters, see:
http://php.net/manual/zh/function.ucfirst.php <http://php.net/manual/zh/function.ucfirst.php>

so~ maybe yii should find some replacement, I see mb_convert_case in the php doc, but it’s not exactly matched.

Thank you for this conversation.

**Comment by cebe** [comment]
I see, thanks for digging into it. I wonder however why this only happens for you and not in my example... will try to fix it anyway.

**Comment by sxcooler** [comment]
Thank you, and I haven’t tested all the situation, different environment causes different result.
Mine: 
PHP 5.6.19,
OSX 10.11.4,
intl lib version 1.1.0 with ICU version 56.1,
intl.default_locale: no value.

system env:
LANG=zh_CN.UTF-8

**Comment by paulhuo** [comment]
Yes,it happened to me too.My system:OSX10.11.5;php 5.5.24
![image](https://cloud.githubusercontent.com/assets/20138852/17355664/d0266114-5984-11e6-931d-095e62145591.png)

**Comment by samdark** [comment]
Just in case, multibyte version of ucfirst...

```php
function mb_ucfirst($string, $encoding = 'UTF-8') {
  $firstChar = mb_strtoupper(mb_substr($string, 0, 1, $encoding), $encoding);
  return $firstChar . mb_substr($string, 1, mb_strlen($string, $encoding), $encoding);
}
```

**Comment by cebe** [comment]
Should be fixed by 08b6c09, please test if this works for you!
